tests/manage/mcg/test_write_to_bucket.py

```diff
@@ -78,3 +104,54 @@ def test_data_reduction(self, mcg_obj, awscli_pod, bucket_factory):
             bucketname = bucket.name
 
         assert mcg_obj.check_data_reduction(bucketname), 'Data reduction did not work as anticipated.'
+
+    @pytest.fixture()
+    def setup_rbd_cephfs_pods(self, multi_pvc_factory, pod_factory):
+        """
+        This fixture setups the required rbd and cephfs pvcs and pods
+
+        """
+        pvc_objs_rbd = multi_pvc_factory(
+            interface=constants.CEPHBLOCKPOOL, size=15, num_of_pvc=8
+        )
+        ns = pvc_objs_rbd[0].project
+
+        pvc_objs_cephfs = multi_pvc_factory(
+            interface=constants.CEPHFILESYSTEM, size=15, num_of_pvc=8, project=ns
+        )
+
+        pods = []
+        for pvc in pvc_objs_rbd:
+            pods.append(pod_factory(
+                interface=constants.CEPHBLOCKPOOL, pvc=pvc)
+            )
+
+        for pvc in pvc_objs_cephfs:
+            pods.append(pod_factory(
+                interface=constants.CEPHFILESYSTEM, pvc=pvc)
+            )
+
+        return pods
+
+    @vsphere_platform_required
+    @pytest.mark.polarion_id("OCS-2040")
+    def test_write_to_bucket_rbd_cephfs(self, setup_rbd_cephfs_pods, retrive_s3_objects,
+                                        mcg_obj, awscli_pod, bucket_factory
+                                        ):
+        """
+        Test RGW restarts after running s3, rbd and cephfs IOs in parallel
+
+        """
+        logger.info('RGW restart count before running IOs')
+        pods = get_rgw_pods()
+        for rgw_pod in pods:
+            rgw_restart_count = rgw_pod.restart_count
+
+        with ThreadPoolExecutor() as p:
+            p.submit(pod_io, setup_rbd_cephfs_pods)
+            p.submit(s3_io, retrive_s3_objects, mcg_obj, awscli_pod, bucket_factory)
+
+        logger.info("Checking whether RGW pod restarted")
+        for rgw_pod in pods:
+            rgw_pod.reload()
+            assert rgw_pod.restart_count == rgw_restart_count, 'RGW pod restarted after running parallel IOs'
```

Comment:
Checking for rgw restart count should become one of the exit criteria for all noobaa based tests. Please make this as a common function and call it to verify the restart count in the applicable tests

Reply:
Added verify rgw pod restart count fixture to bucket creation and deletion tests. Mainly intended for performance tests which does 1000 bucket creation and deletion.